backend/Api/Projects/ProjectController.cs

```diff
@@ -19,33 +19,32 @@ public class ProjectController : ControllerBase
 
     private readonly IMemoryCache _cache;
     private readonly ApplicationContext _context;
+    private readonly IEngagementRepository _engagementRepository;
+
+    private readonly IOrganisationRepository _organisationRepository;
 
-    public ProjectController(ApplicationContext context, IMemoryCache cache)
+    public ProjectController(ApplicationContext context, IMemoryCache cache,
+        IOrganisationRepository organisationRepository, IEngagementRepository engagementRepository)
     {
         _context = context;
         _cache = cache;
+        _organisationRepository = organisationRepository;
+        _engagementRepository = engagementRepository;
     }
 
     [HttpGet]
     [Route("get/{projectId}")]
-    public ActionResult<ProjectWithCustomerModel> GetProject([FromRoute] string orgUrlKey,
-        [FromRoute] int projectId)
+    public async Task<ActionResult<ProjectWithCustomerModel>> GetProject([FromRoute] string orgUrlKey,
+        [FromRoute] int projectId, CancellationToken ct)
     {
-        var service = new StorageService(_cache, _context);
-
-        var selectedOrg = _context.Organization.SingleOrDefault(org => org.UrlKey == orgUrlKey);
+        var selectedOrg = await _organisationRepository.GetOrganizationByUrlKey(orgUrlKey, ct);
         if (selectedOrg is null) return BadRequest("Selected org not found");
 
-        var project = service.GetProjectWithCustumerById(projectId);
-
-        if (project is null)
-        {
-            return NoContent();
-        }
+        var engagement = await _engagementRepository.GetEngagementById(projectId, ct);
 
-        var responseModel =
-            new ProjectWithCustomerModel(project.Name, project.Customer.Name, project.State, project.IsBillable, project.Id);
+        if (engagement is null) return NoContent();
```

Comment:
[404](https://http.cat/status/404) er hvis du ikke finner noe eller det ikke eksisterer, [204](https://http.cat/status/204) er hvis tingen var vellykket men det er ingenting klienten trenger å gjøre, f.eks. hvis du sletter noe  
```suggestion
        if (engagement is null) return NotFound();
```